From 5f078c7777e6d61971651faa57e772d2f73bb5ee Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 6 Dec 2025 20:04:42 +0800
Subject: [PATCH 084/133] std.builtin: reorder CallingConvention to keep in
 sync with upstream 0.15.2 compiler
X-Developer-Signature: v=1; a=openpgp-sha256; l=954; i=xtex@astrafall.org;
 h=from:subject; bh=Ac5rCnVk9Wpoe9BnQQv3AQy2Jkgv04Y1+y1eBrZ3d1M=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78tjntUcd8XE9nlHdl1iR2/Ds21eTK5qP7ZJEz4kzn
 PvkEv+gYyILgxgXg6WYIkuRYYM3q046v+iyclmYOaxMIEOkRRoYGBgYWBj4chPzSo10jPRMtQ31
 DA11jHSMGLg4BWCq+d0ZGa5N3nb7cyOf5323P0cFNOfo/plx5R/PEoki4T2rqtu4O1MZGeZ3Tl5
 awmS6LuvrtW//yywX6fQm+K78sVZm9swumdN9EYwA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Change-Id: Idc1df5a495eed2a128510040b25fdf49881acdd1
---
 lib/std/builtin.zig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/std/builtin.zig b/lib/std/builtin.zig
index 4240cb42f9ad..d5c8891e187c 100644
--- a/lib/std/builtin.zig
+++ b/lib/std/builtin.zig
@@ -207,7 +207,6 @@ pub const CallingConvention = union(enum(u8)) {
 
     // Calling conventions for the `x86_64` architecture.
     x86_64_sysv: CommonOptions,
-    x86_64_x32: CommonOptions,
     x86_64_win: CommonOptions,
     x86_64_regcall_v3_sysv: CommonOptions,
     x86_64_regcall_v4_win: CommonOptions,
@@ -353,6 +352,8 @@ pub const CallingConvention = union(enum(u8)) {
     spirv_fragment,
     spirv_vertex,
 
+    x86_64_x32: CommonOptions,
+
     /// Options shared across most calling conventions.
     pub const CommonOptions = struct {
         /// The boundary the stack is aligned to when the function is called.
-- 
2.52.0

